core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1145,6 +1162,113 @@ public List<FileInfo> listStatus(AlluxioURI path, ListStatusContext context)
     return fileInfos;
   }
 
+  /**
+   * If performing a partial listing, from an offset, this checks to ensure the offset
+   * exists within the starting path.
+   * @param path the listing path
+   * @param context the listing context
+   * @return the components of the path to the offset
+   */
+  private @Nullable List<String> checkPartialListingOffset(
+      AlluxioURI path, ListStatusContext context)
+      throws FileDoesNotExistException, InvalidPathException {
+    List<String> partialPathNames = null; // null if we are not using a partial listing
```

Comment:
Can we just use an empty list instead of null? That way we can save the annotation and nobody needs to worry about nulls. Empty list in many cases don't even need special handling.